### PR TITLE
Added backend readme

### LINKED ISSUE #111: Update the Readme.md

### PR DESCRIPTION
**Pull request by 17himanshu**
fixes #111

### CODE REVIEW

**Comment by KS963000** [comment]
Include Real-time Collaboration in the list of features, and make sure to add a /wss route for websockets.

File: backend/controllers/userController.js
Comment (ravinder-chadha) [inline backend/controllers/userController.js:1]: Why is this getting included in backend commit?

**Review by ravinder-chadha — APPROVED** [review]
Looks fine

File: backend/controllers/userController.js [hunk 1]
```diff
@@ -1,7 +1,6 @@
 const bcrypt = require('bcryptjs');
 const jwt = require('jsonwebtoken');
 const User = require('../models/userModel');
-const CustomError = require("../utils/CustomError.js");
```
Comment (17himanshu) [inline backend/controllers/userController.js:4]: Sir, while resolving merge conflicts, this import of CustomError was made redundant and caused an error. So with this pr, I removed it. That's why I have added this file intentionally

**Review by ravinder-chadha — APPROVED** [review]

**Pull request merged**